packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -296,50 +297,75 @@ const helpers = {
 
   negotiateOptionsForInvalidRequest(
     httpResponses: IHttpOperationResponse[],
-    statusCodes: NonEmptyArray<number>
+    statusCodes: NonEmptyArray<number>,
+    exampleKey?: string
   ): RE.ReaderEither<Logger, Error, IHttpNegotiationResult> {
+    const buildResponseBySchema = (response: IHttpOperationResponse, logger: Logger) => {
+      logger.debug(`Unable to find a content with an example defined for the response ${response.code}`);
+      // find first response with a schema
+      const responseWithSchema = response.contents && response.contents.find(content => !!content.schema);
+      if (responseWithSchema) {
+        logger.success(`The response ${response.code} has a schema. I'll keep going with this one`);
+        return E.right({
+          code: response.code,
+          mediaType: responseWithSchema.mediaType,
+          schema: responseWithSchema.schema,
+          headers: response.headers || [],
+        });
+      } else {
+        return pipe(
+          createEmptyResponse(response.code, response.headers || [], ['*/*']),
+          E.fromOption(() => {
+            logger.debug(`Unable to find a content with a schema defined for the response ${response.code}`);
+
+            return new Error(`Neither schema nor example defined for ${response.code} response.`);
+          })
+        );
+      }
+    };
+
+    const buildResponseByExamples = (
+      response: IHttpOperationResponse,
+      contentWithExamples: IWithExampleMediaContent,
+      logger: Logger
+    ) => {
+      logger.success(`The response ${response.code} has an example. I'll keep going with this one`);
+      return pipe(
+        O.fromNullable(exampleKey),
+        O.fold(
+          () => O.fromNullable(contentWithExamples.examples[0]), // if exampleKey is not specified use first example
+          exampleKey => findExampleByKey(contentWithExamples, exampleKey)
+        ),
+        O.fold(
+          () => {
+            throw new Error(`An example with the specified exampleKey does not exist for ${response.code} response.`);
```

Comment:
I don't think we should throw an error here. It causes Prism to not return a response because there's an unhandled promise rejection
![image](https://user-images.githubusercontent.com/33431856/115607582-c4832080-a2aa-11eb-82ce-7aac41a29a56.png)

In this case since the person requested an example that does not exist for the code, I think we should:
1. log out that the example they requested does not exist in the spec for the code being returned
2. fall back to the first example defined in the spec

if they have no examples defined in the spec then this path doesn't apply, and it'll generate a response based off of the schema for the response

Reply:
Yeah make, sense. Did an update.

```sh
 curl --location --request POST 'http://127.0.0.1:4010/api/users/register' \
--header "Content-Type: application/json" \
--header 'Prefer: example=missing' \
--data '{"username":"xyz","password":"xyz", "email": "john@example.com"}'
{"message":"The username is too short"}%

[12:47:07 PM] › [HTTP SERVER] post /api/users/register ℹ  info      Request received
[12:47:07 PM] ›     [NEGOTIATOR] ℹ  info      Request contains an accept header: */*
[12:47:07 PM] ›     [VALIDATOR] ⚠  warning   Request did not pass the validation rules
[12:47:07 PM] ›     [NEGOTIATOR] ✔  success   Found response 422. I'll try with it.
[12:47:07 PM] ›     [NEGOTIATOR] ✔  success   The response 422 has an example. I'll keep going with this one
[12:47:07 PM] ›     [NEGOTIATOR] ⚠  warning   An example with key missing does not exist. The first example will be used.
[12:47:07 PM] ›     [NEGOTIATOR] ✔  success   Responding with the requested status code 422
[12:47:07 PM] ›     [VALIDATOR] ✖  error     Violation: request.body.password should NOT be shorter than 8 characters

```

Reply:
What is does prism do normally when an example isn't found for an successful(200) response? Does it default to a random one, or pick one randomly if there are multiple examples.

I was reading the docs and it wasn't clear to me what happens.

Reply:
you could play around with it and find out 😄 .

Reply:
I could, and would and will. Just curious if anyone knows.